.vscode/launch.json

```diff
@@ -29,6 +29,15 @@
 			"skipFiles": ["<node_internals>/**"],
 			"type": "node",
 		},
+		{
+			"name": "npm test build-infrastructure",
+			"cwd": "${workspaceFolder}/build-tools/packages/build-infrastructure",
+			"request": "launch",
+			"runtimeArgs": ["run-script", "test"],
+			"runtimeExecutable": "npm",
+			"skipFiles": ["<node_internals>/**"],
+			"type": "node",
+		},
```

Comment:
Is there a way to run tests under the debugger other than this?

Reply:
Yes, in package.json hover over the npm script you want to run and click "Debug script", which will open a JavaScript Debug Terminal in VSCode and launch it from there, so the debugger auto-attaches. Manually opening a JS debug terminal and launching any process has the same effect.

![image](https://github.com/user-attachments/assets/ba29aebc-03d4-4831-840d-d2c2520146c4)

Reply:
Nice, I didn't know about that. One drawback is that it doesn't seem to work with F5 and related shortcuts to run. But maybe there is a shortcut I can assign to "re-run most recently debugged test" or something.

Reply:
I wouldn't oppose having a VSCode workspace for build-tools [like we have for tree](https://github.com/microsoft/FluidFramework/tree/main/packages/dds/tree/.vscode) and adding custom launch configurations there. Just putting them at the root is what bothers me a bit, since the list is already pretty long and the new entry only applies to a very specific part of the repo structure.